### DISCUSSION TITLE
Validation of action summary - possibly from call service node secondary output

### DISCUSSION THREAD

**Discussion by jaburges** [zachowj/node-red-contrib-home-assistant-websocket#690] [Feature Request]
How do you find out if actions were not completed in a long branching flow?

Thinking out loud, the debug node would capture outputs of the node action, but not the device/entity response success/failure, which you could hold and assemble, but I was wondering when you have long flows, and maybe an MQTT light, Zwave switch or a tuya device just doesnt respond - is there a way to report back.

Scenario:
"Hey Google, Turn on bedtime" ....(turns on routine).....responds with "Bedtime completed, but downstairs door did not lock"

Now this is possible with hugely complex flows (using the current state node), providing a different response for a door lock, or light group etc

BUT

What if a 'call service' node could confirm an entity responded correctly? (rather than just the action was performed successfully)?
I guess the challenge is that certain devices like locks, take longer to confirm, so you'd need a hold until X node at the end to gather the msgs. 

Could a 'call service' node have a secondary output, that waits until the action is completed or for a specified time out and sends it to a "reporting" branch of the flow to compile all the messages to be summarized?

**Comment by zachowj** [comment]
I think something like this would over-complicate the call-service node. I could see someone building a subflow that could replace a call-service node and being able to do all that you want without modifying the call-service node.

**Reply by jaburges** [reply]
fair point. Interesting idea - so you create a subflow and call the same subflow from output. 
Then i guess you'd have to wait for an approx amount of time to gather all the outputs and compile a message.

**Comment by bthome** [comment]
Is it possible to achieve some of this?  Collect the state of the entity before and after of the service call?  If on an event I take an action (service call) but I want to know what the state of the entity was before and after.  I'm sure I can add more nodes to get state, but I want to combine them to notify state went from Off -> On.
